### PR TITLE
happy-thoughts-api-Therese-Linden

### PR DESCRIPTION
**Pull request by thereselinden**
Happy that I managed to do infinite scroll for this project. 

**Heroku:** 
https://happy-thoughts.herokuapp.com/

**Front-end:** 
https://github.com/thereselinden/project-happy-thoughts

**Netlify:** 
https://happy-thoughts-therese-linden.netlify.app/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -7,23 +7,116 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
+    type: Number,
+    default: 0
+  },
+  name: {
+    type: String,
+    default: "Anonymous"
+  },
+  createdAt: {
+    type: Date,
+    default: () => new Date()
```
Comment (sarahejhej) [inline server.js:28]: Great with the defensive coding here!

File: server.js [hunk 2]
```diff
@@ -7,23 +7,116 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
+//___________________Middleweares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
+const listEndpoints = require('express-list-endpoints')
+
+//___________________Error messages
+const SERVICE_UNAVAILABLE = "service unavailable"
+const BAD_REQUEST = "Bad request"
+const POST_THOUGHT_FAILED = "Could not save thought"
+const ID_NOT_FOUND = "Thought ID was not found"
```
Comment (sarahejhej) [inline server.js:45]: Keeps it more simple with all error messages in the same place 👍

File: server.js [hunk 3]
```diff
@@ -7,23 +7,116 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+//___________________GET all thoughts
+app.get('/thoughts', async (req, res) => {
+  const { sort, page } = req.query
+
+  const pageNumber = +page || 1;
```
Comment (sarahejhej) [inline server.js:67]: We worked together, so we faced the same problems. Great that we managed to solve the infinite scroll!

File: server.js [hunk 4]
```diff
@@ -7,23 +7,116 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
+  } else {
+    res.status(400).send({ error: BAD_REQUEST, error: err.errors })
+  }
+})
+
+//___________________POST thought
+app.post('/thoughts', async (req, res) => {
+  const { name, message } = req.body
+  try {
+    const thought = await new Thought({ message: message, name: name}).save()
```
Comment (sarahejhej) [inline server.js:101]: This could probably be written as just `{message, name}` since the fields are named the same

**Review by sarahejhej — APPROVED** [review]
Great work Therese! 
Since we worked together, I don't have much to add. You're consistent in not using semicolons and with indentation ⭐️ Your comments are precise and easily spotted. I'm happy that we managed to solve the infinite scroll, it was a tricky one.
Well done! 🏅

**Review by KarolCislowski — APPROVED** [review]
Awesome Project 🥇 🥇

**Pull request closed without merging**